server/core/src/rodsServer.cpp

```diff
@@ -1306,7 +1343,39 @@ int initServerMain(
 
     if (reServerHost && LOCAL_HOST == reServerHost->localFlag) {
         rodsLog(LOG_NOTICE, "Forking Rule Execution Server (irodsReServer) ...");
+        ix::cron::cron_builder delay_server;
+        delay_server.to_execute([&](std::any& _){
+            auto delay_pid = irods::get_server_property<int>(irods::RE_PID_KW);
+            if(waitpid(delay_pid,nullptr,WNOHANG)!=0){
+                ix::log::server::info("Restarting rule server");
+                const int pid = RODS_FORK();
+                if (pid == 0) {
+                    close(svrComm->sock);
+
+                    std::vector<char*> argv;
+                    argv.push_back("irodsReServer");
+
+                    if (enable_test_mode) {
+                        argv.push_back("-t");
+                    }
+
+                    if (write_to_stdout) {
+                        argv.push_back("-u");
+                    }
+
+                    argv.push_back(nullptr);
+
+                    // Launch the delay server!
+                    execv(argv[0], &argv[0]);
+                    exit(1);
+                }
+                else {
+                    irods::set_server_property<int>(irods::RE_PID_KW, pid);
+                }
+            }
```

Comment:
wrap in a function to avoid code duplication

Reply:
:+1: